### PR TITLE
dom-api-task

### PR DESCRIPTION
**Pull request by vl-shevchenko**
###dom-api-task

[Demo](https://vl-shevchenko.github.io/dom/) |
[Code base](https://github.com/vl-shevchenko/dom)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by vl-shevchenko** [comment]
Hey!
My code has been validated.
I took into account all the tips from common mistakes when writing code.

File: submissions/vl-shevchenko/dom-api-task/script.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+const city = [
+    {
+        color: 'url(images/kyiv5-904w.jpg)',
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/script.js:3]: Colour is a link to image? Should have another name. And remove `url`.

File: submissions/vl-shevchenko/dom-api-task/script.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    },
+    {
+        color: 'url(images/Odesa.jpg)',
+        name: 'Odesa',
+        info: 'City by the Sea, the birthplace of many famous Ukrainian immigrants and the capital of humor. Here is a special atmosphere, special people and special attitude to everything, you need to feel it. In early April Odessa helds the festival of humor.',
+    },
+];
+
+let defaultContent = document.querySelector('.container');
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/script.js:35]: If you do not change a variable(with `=` operator) always use const.

File: submissions/vl-shevchenko/dom-api-task/script.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+defaultContent.innerHTML = `<p class='info'>${city[0].info}</p>`;
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/script.js:36]: Should be in function for setting default content.

File: submissions/vl-shevchenko/dom-api-task/script.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+document.body.style.backgroundImage = city[0].color;
+let defaultName = document.querySelector('.name-city');
+defaultName.innerText = `${city[0].name}`;
+
+const buildMenu = city => {
+    document.querySelector('.nav__bloc').innerHTML = `<ul class='container-list'></ul>`;
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/script.js:42]: `.nav__block` functional and `city` should be in different functions.

File: submissions/vl-shevchenko/dom-api-task/script.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    city.map((val) => {
+        let nameCity = document.createElement('li');
+        nameCity.classList.add('menu-list');
+        nameCity.innerHTML = `<a href="#" class='menu-link'>${val.name}</a>`;
+
+        document.querySelector('.container-list').appendChild(nameCity);
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/script.js:48]: You already have var for `.container-list` on line [53](https://github.com/kottans/frontend-2022-homeworks/pull/426/files#diff-67c3e77f177ce6616994c08bfed32b00d13bc21085c15f08c825ebb6e7cfb2eeR53). Why you reselect it?

File: submissions/vl-shevchenko/dom-api-task/style.css [hunk 6]
```diff
@@ -0,0 +1,278 @@
+html {
+    box-sizing: border-box;
+    font-size: 16px;
+
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/style.css:4]: Remove redundant empty lines.

File: submissions/vl-shevchenko/dom-api-task/script.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+};
+buildMenu(city);
+
+const menuButton = document.querySelector('.container-list');
+menuButton.addEventListener('click', function (e) {
+    document.querySelector('.container').innerHTML = '';
+
+    let currentCity = e.target.innerHTML;
+    let info = '';
+    let color = '';
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/script.js:59]: It is not a color.

File: submissions/vl-shevchenko/dom-api-task/script.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    let name = '';
+    for (let i = 0; i < city.length; i++) {
+        if (city[i].name == currentCity) {
+            info = city[i].info;
+            color = city[i].color;
+            name = city[i].name;
+        }
+    }
+
+    document.body.style.backgroundImage = color;
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/script.js:69]: Remove temp `let` vars, move this logic to `if` at line [62](https://github.com/kottans/frontend-2022-homeworks/pull/426/files#diff-67c3e77f177ce6616994c08bfed32b00d13bc21085c15f08c825ebb6e7cfb2eeR53) and move `if` body to function.

File: submissions/vl-shevchenko/dom-api-task/script.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    citiesName.innerHTML = `${name}`;
+    let citiesInfo = document.querySelector('.container');
+    citiesInfo.innerHTML = `<p class='info'>${info}</p>`;
+});
+
+const burger = document.querySelector('#burger');
+const popup = document.querySelector('#popup');
+burger.addEventListener('click', burgerhandler);
+function burgerhandler(e) {
+    e.preventDefault();
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/script.js:80]: Why do we need here `preventDefault`?

File: submissions/vl-shevchenko/dom-api-task/style.css [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+    padding: 2% 0 0 0;
+}
+
+.nav__bloc {
+    display: flex;
+    justify-content: center;
+    width: 250px;
+    min-height: 300px;
+    height: 80%;
+    background: rgba(255, 237, 224, 0.56);
```
Comment (lysenko-sergey-developer) [inline submissions/vl-shevchenko/dom-api-task/style.css:65]: Move colors to css3 vars.

**Comment by lysenko-sergey-developer** [comment]
@vl-shevchenko Nice work! 💯

**Review by lysenko-sergey-developer — CHANGES_REQUESTED** [review]

**Comment by lysenko-sergey-developer** [comment]
UI/UX Improvements:
Research more about [Glassmorphism](https://www.freecodecamp.org/news/glassmorphism-design-effect-with-html-css/). Try to make your texts more readable.

**Pull request closed without merging**